sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,261 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.github.sevntu.checkstyle.Utils;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Check that forbidden methods are not used. We can forbid methods by ruleName and number of
+ * arguments.
+ * This can be used to enforce things like:
+ * <ul>
+ * <li> System.exit() should not be called.</li>
+ * <li> Assert.assertTrue() and Assert.assertFalse() have a 1 arg variant that does not provide a
+ * helpful message on failure.
+ * </ul>
+ * Parameters are:
+ * <ul>
+ * <li><b>methodName</b> - Regex to match name of the method to be forbidden.
+ * When blank or unspecified, all the methods are allowed.</li>
+ * <li><b>argumentCount</b> - Range to match number of arguments the method takes.
+ * When blank or unspecified, only method name will be used for check.
+ * </ul>
+ *
+ * <p>An example configuration:
+ * <pre>
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="exit"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="assert(True|False)"/&gt;
+ *    &lt;property name="argumentCount" value="1"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="assertEquals"/&gt;
+ *    &lt;property name="argumentCount" value="2"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * Argument count can be bounded range (e.g.: {@code 2-4}) or unbounded range
+ * (e.g.: {@code -5, 6-}). Unbounded range can be unbounded only on one side.
+ * Multiple ranges must be comma separated.
+ * For example, the following will allow only 4 and 8 arguments.
+ *
+ * <pre>
+ * &lt;module name="ForbidCertainMethodsCheck"&gt;
+ *    &lt;property name="methodName" value="asList"/&gt;
+ *    &lt;property name="argumentCount" value="-3, 5-7, 9-"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * <p>Note: Because of limitations of how checkstyle works we can't check type of the
+ * arguments.
```

Comment:
Please remove this note, it is well known limitation.

Reply:
Done.